doc/spec_TransferElement.dox

```diff
@@ -239,7 +243,8 @@ This documnent is currently still **INCOMPLETE**!
 - 8.5.2 As the asynchonous mechanism and a synchronous read are two idependent channels there are potential race condition, depending on the exact protocol.
   The backend has to avoid this if possible. If it cannot be avoided, the implementation must make sure that the last value in the queue is the newest value, and this is not dopped or missed, even if the values before are not in order or send twice.
 
-- 9. It does not matter if the exception occured in an asynchronous or synchronous read, or in a write operation.
+- \anchor transferElement_comment_B_9a \ref transferElement_B_9 "9." The ChimeraTK::runtime_error is the only exception that conceptually is recoverable by calling DeviceBackend::open(). ChimeraTK::logic_error is a programming or configuration mistake. The boost::numeric::bad_numeric_cast conceptually also is a logical error (wrong data type) which only shows at run time when the data is overflowing. The boost::thread_interrupted is thrown when TransferElement::interrupt() is called. It means that a blocking read() has been interrupted on request, not that the device is malfunctioning and would need recovery.
```

Comment:
That to me sounds that only runtime_error is recoverable form only by open(), the others by other means not further specified.
```
ChimeraTK::runtime_error is the only exception that conceptually is recoverable. This is done by calling DeviceBackend::open(). Other exceptions we expect are:
 - ctk::logic_error which is a programming or configuration error and cannot be recovered
 - boost::numeric_bad_numeric_cast which is conceptually identical to a logic error (wrong data type). It only shows at runtime when the data is overflowing
 - boost::thread_interrupted is used to cleanly shut down threads. It is not an error, but a way to interrupt a blocking read() on request, by calling TransferElement::interrupt()
```
The explanation of boost::thread_interrupted might need a bit of more refinement

Reply:
> boost::thread_interrupted is used to cleanly shut down threads.
That's what it can be used for. But it can also be used do just cancel an read request. For instance you wait for a variable while a calibration is running. If the calibration succeeds without any thing being read, you don't need the value any more. You call interrupt() on that read request and continue with something else.
I would not put what we use it for into the spec.

Reply:
> It is not an error, but a way to interrupt a blocking read().
No, the exception is not the way to interrupt the read. The call to TE::interrupt is.
I also would not use the tern "error" in this case. Errors in the sense of logical errors / programming mistakes are conceptually not recoverable. The point here that it is not a "fault", i.e. malfunctioning which could be recovered.

Reply:
Fine with me. The main point was that it sounded as if those were faults recoverable by other means than open()

Reply:
Ah. that sentence I can put in. And maybe do the bullet points for better overview.